mtk_logger.cc

```diff
@@ -612,14 +609,15 @@ static void mtk_read()
   dbg(1, "Download %dkB from device\n", (addr_max+1) >> 10);
 
   if (dsize > addr_max) {
-    dbg(1, "Temp %s file (%ld) is larger than data size %d. Data erased since last download !\n", TEMP_DATA_BIN, dsize, addr_max);
+    dbg(1, "Temp %s file (%ld) is larger than data size %d. Data erased since last download !\n", qPrintable(TEMP_DATA_BIN), dsize, addr_max);
     fclose(dout);
     dsize = 0;
     init_scan = 0;
-    rename(TEMP_DATA_BIN, TEMP_DATA_BIN_OLD);
-    dout = fopen(TEMP_DATA_BIN, "wb");
+    QFile::rename(TEMP_DATA_BIN, TEMP_DATA_BIN_OLD);
+    dout = ufopen(TEMP_DATA_BIN, "wb");
```

Comment:
QFile::rename will fail if TEMP_DATA_BIN_OLD exists without overwriting it.  The behavior of rename may be implementation dependent.  See https://wiki.sei.cmu.edu/confluence/display/c/FIO10-C.+Take+care+when+using+the+rename%28%29+function